examples/html-kitchen-sink/stories/index.stories.js

```diff
@@ -0,0 +1,38 @@
+import { document } from 'global';
+import { storiesOf } from '@storybook/html';
+import { action } from '@storybook/addon-actions';
+import { linkTo } from '@storybook/addon-links';
+import { subscriptionsStore } from '@storybook/core/client';
+
+import './welcome.css';
+import welcome from './welcome.html';
+
+// TODO move to addon-links
+const linkSubscribtion = () => {
+  const listener = e => {
+    const { sbKind, sbStory } = e.target.dataset;
+    if (sbKind || sbStory) {
+      e.preventDefault();
+      linkTo(sbKind, sbStory)();
+    }
+  };
+
+  document.addEventListener('click', listener);
+  return () => document.removeEventListener('click', listener);
+};
+
+storiesOf('Welcome', module)
+  .addDecorator(story => {
+    subscriptionsStore.register(linkSubscribtion);
```

Comment:
I don't really understand the idea of using a subscription+decorator to make this run each time we render a story. Why not just register the global event handler once in `addon-links`?

Reply:
> Isn't that what the decorator essentially does though

Nope, it only subscribes the first time, subsequently it only marks subscription as used so that it's not removed after render

> I don't really get what the subscriptionStore has to do with this PR

We need to support addons like knobs which use a lot of subscriptions. In other frameworks we use lifecycle hooks for that, e.g.: https://github.com/storybooks/storybook/blob/master/addons/knobs/src/react/WrapStory.js#L18-L37

The `subscriptionStore` is a framework-agnostic solution for this, so we can:

a) Use such addons in @storybook/html (that's why it's this PR)
b) Remove a lot of existing framework specifics from addons

Reply:
Awesome!